core/src/main/scala/caliban/execution/Executor.scala

```diff
@@ -394,7 +384,19 @@ object Executor {
           wrappers match {
             case Nil             => query
             case wrapper :: tail =>
-              val q = if (isPure && !wrapper.wrapPureValues) query else wrapper.wrap(query, fieldInfo)
+              val q =
+                if (isPure && !wrapper.wrapPureValues) query
+                else
+                  wrapper
+                    .wrap(query, fieldInfo)
+                    .mapErrorCause(e =>
+                      effectfulExecutionError(
+                        PathValue.Key(fieldInfo.name) :: fieldInfo.path,
+                        Some(fieldInfo.details.locationInfo),
+                        e
+                      )
+                    )
+
```

Comment:
I think we should re-write the entire `wrap` method to something like this, to avoid applying the error handler on every field wrapper:

```scala
      def wrap(query: ExecutionQuery[ResponseValue], isPure: Boolean, fieldInfo: FieldInfo) = {
        @tailrec
        def loop(query: ExecutionQuery[ResponseValue], wrappers: List[FieldWrapper[R]]): ExecutionQuery[ResponseValue] =
          wrappers match {
            case Nil             => query
            case wrapper :: tail =>
              val q =
                if (isPure && !wrapper.wrapPureValues) query
                else wrapper.wrap(query, fieldInfo)
              loop(q, tail)
          }

        if ((isPure && !wrapPureValues) || (fieldWrappers eq Nil)) query
        else {
          loop(query, fieldWrappers).mapErrorCause(e =>
            effectfulExecutionError(
              PathValue.Key(fieldInfo.name) :: fieldInfo.path,
              Some(fieldInfo.details.locationInfo),
              e
            )
          )
        }
      }
```

What do you think?

Reply:
Good point!